engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -895,8 +895,10 @@ private void redoPartitions(PMapping currentMapping, String note)
         addedPartitions.add(newPartition);
       } else {
         // check whether mapping was changed
+        int currentPartitionsSize = mainPC.currentPartitions.size();
         for (DefaultPartition<Operator> pi : mainPC.currentPartitions) {
-          if (pi == newPartition && pi.isModified()) {
+          if (pi == newPartition && (pi.isModified() ||
```

Comment:
@tweise Should not redeploy of downstream being done only when downstream pipeline has a unifier and only unifier needs to be redeployed for the pipeline connected to the current partition? For example, if logical dag is defined as:
```
A->B-C->Aggregator
```
and it is partitioned to 2 pipelines starting with A:
```
A1->B1->C1->
             U->Aggregator
A2->B2->C2->
```
and A is repartitioned to 3:
```
A1->B1->C1->
A2->B2->C2->U->Aggregator
A3->B3->C3->
```
A1->B1->C1 and A2->B2->C2 don't need to be redeploed?